articles/rules/preview/index.md

```diff
@@ -0,0 +1,314 @@
+</div>
+
+
+### Add roles to a user
+
+In this example, all authenticated users will get a **guest** role, but `johnfoo@gmail.com` will also be an **admin**:
+
+```js
+function (user, context, callback) {
+  user.roles = [];
+  // only johnfoo is admin
+  if (user.email === 'johnfoo@gmail.com') {
+    user.roles.push('admin');
+  }
+
+  // all users are guest
+  user.roles.push('guest');
+
+  callback(null, user, context);
+}
+```
+
+At the beginning of the rules pipeline, John's `user` object will be:
+
+```js
+{
+  "email": "johnfoo@gmail.com",
+  "family_name": "Foo",
+  "user_id": "google-oauth2|103547991597142817347"
+  //... other properties ...
+}
+```
+
+The `context` object will be:
+
+```js
+{
+  "clientID": "...client_id_of_the_app...",
+  "clientName": "my app",
+  "clientMetadata": {
+    "myKey1": "myValue2",
+    "myKey2": "myValue2"
+  }
+  "connection": "google-oauth2"
+}
+```
+
+After the rule executes, the output that the application will receive is the following `user` object:
+
+```js
+{
+  "email": "johnfoo@gmail.com",
+  "family_name": "Foo",
+  "user_id": "google-oauth2|103547991597142817347",
+
+  //... other props ...
+
+  "roles": ["guest", "admin"]  // NEW PROPERTY ADDED BY THE RULE
+}
+```
+
+<div class="alert alert-info">
+Properties added in a rule are <strong>not persisted</strong> in the Auth0 user store. Persisting properties requires calling the Auth0 Management API.
+</div>
+
+### Deny access based on a condition
+
+In addition to adding and removing properties from the user object, you can return an *access denied* error.
+
+```js
+function (user, context, callback) {
+  if (user.roles.indexOf('admin') === -1) {
+    return callback(new UnauthorizedError('Only admins can use this'));
+  }
+
+  callback(null, user, context);
+}
+```
+
+This will cause a redirect to your callback url with an `error` querystring parameter containing the message you set. (e.g.: `https://yourapp.com/callback?error=unauthorized&error_description=Only%20admins%20can%20use%20this`). Make sure to call the callback with an instance of `UnauthorizedError` (not `Error`).
+
+<div class="alert alert-info">
+  Error reporting to the app depends on the protocol. OpenID Connect apps will receive the error in the querystring. SAML apps will receive the error in a <code>SAMLResponse</code>.
+</div>
+
+### API Authorization: Modify Scope
+
+This will override the returned scopes of the `access_token`. The rule will run after user authentication and before authorization.
+
+```js
+function(user, context, callback) {
+
+  // change scope
+  context.accessToken.scope = ['array', 'of', 'strings'];
+
+  callback(null, user, context);
+}
+```
+
+The user will be granted three scopes: `array`, `of`, and `strings`.
+
+### API Authorization: Add Claims
+
+This will add one custom namespaced claim at the `access_token`, and two at the `id_token`.
```

Comment:
this example is great, and is the general direction we are taking on the new pipeline. you no longer modify the "user" object and that auto magically gets to the app. now the power is on the rule. if you want something on the id_token you need to explicitly code for it on a rule by adding claims to the id_token/access_token. this is important because no longer every client gets a full dump of the data of the user, but the authorization server (via the rules) can say "to this client this information"